Vostok.ClusterClient.Transport.Tests/Helpers/SlowStream.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Vostok.Clusterclient.Transport.Tests.Helpers
+{
+    public class SlowStream : MemoryStream
+    {
+        public SlowStream(byte[] data) : base(data)
+        {
+            
+        }
+        
+        public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            await Task.Delay(TimeSpan.FromSeconds(2), cancellationToken).ConfigureAwait(false);
```

Comment:
I think we can do better.
We create SlowStream and CloseUnderliningConnection together. So we can perfectly synchronize them.
For example, using System.Threading.Barrier or even TaskCompletionCourse allows you to wait for another thread to reach the sync point. And it's guaranteed to continue working after that.

Reply:
It seems to me that `AutoResetEvent` is a bit more convenient, as we're waiting for some event to happen so to continue with reading. Also, it's a bit easier to handle state as it's possible to set initial state.